flink-table-store-core/src/main/java/org/apache/flink/table/store/file/mergetree/MergeTreeWriter.java

```diff
@@ -163,7 +163,11 @@ public void flushMemory() throws Exception {
                 // adding one record then remove one record, but after merging this record will not
                 // appear in lsm file. This is OK because we can also skip this changelog.
                 DataFileMeta fileMeta = writer.result();
-                if (fileMeta != null) {
+                if (fileMeta == null) {
+                    for (String extraFile : extraFiles) {
```

Comment:
Can we add a test for this? Maybe in `ChangelogWithKeyFileStoreTableTest.testStreamingChangelog`